libdevice/imf_bf16.hpp

```diff
@@ -81,6 +81,85 @@ __float2bfloat16(float f, __iml_rounding_mode rounding_mode) {
 
   return (bf16_sign << 15) | (bf16_exp << 7) | bf16_mant;
 }
+// We only need utils to convert double to bfloat16 with RTE
+static _iml_bf16_internal __double2bfloat16(double d) {
+  union {
+    double d_val;
+    uint64_t u64_val;
+  } fp64_bits;
+
+  fp64_bits.d_val = d;
+  uint16_t bf16_sign =
+      static_cast<uint16_t>((fp64_bits.u64_val & 0x8000000000000000) >> 63);
+  uint16_t fp64_exp =
```

Comment:
I believe here we can do int16_t and get cast at conversion point with bias.

Reply:
Hi, @zettai-reido 
I changed the 'uint16_t' to 'int16_t' and removed the unnecessary 'static_cast', it looks like cast is not needed since bf16_sign will be 0 or 1 and (bf16_sign << 16) should be OK without casting to uint16_t.
Thanks very much.

Reply:
@jinge90  I thought about `fp64_exp`.
For it signed int16_t makes more sense.
What do you think?